envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -0,0 +1,155 @@
+//
+//     kind: ServiceRole
+//     metadata:
+//       name: products-viewer
+//       namespace: default
+//     spec:
+//       rules:
+//       - services: ["products.default.svc.cluster.local"]
+//         paths: ["/products/*"]
+//         verbs: ["GET", "HEAD"]
+//         constraints:
+//         - key: request.http.headers["version"]
+//           values: ["v1", "v2"]
+//
+// A ServiceRoleBinding specification includes two parts:
+// * "roleRef" refers to a ServiceRole object.
+// * A list of "subjects" that are assigned the roles.
+//
+// A subject can be a "user" or a "group", or represented with a set of attributes. Specifically,
+// "user" is mapped to "source.principal" attribute. "group" is currently undefined. It is reserved
+// to be used in the future.
+//
+// Below is an example of ServiceRoleBinding object "test-binding-products", which binds two
+// subjects to ServiceRole "product-viewer":
+//   * User (service account) "cluster.local/ns/default/sa/bookinfo-reviews"
+//   * Service "reviews.default.svc.cluster.local" at version "v1".
+//
+//     kind: ServiceRoleBinding
+//     metadata:
+//       name: test-binding-products
+//       namespace: default
+//     spec:
+//       subjects:
+//       - user: cluster.local/ns/default/sa/bookinfo-reviews
+//       - attributes:
+//           source.service: "reviews.default.svc.cluster.local"
+//           source.labels["version"]: "v1"
+//       roleRef:
+//         name: "products-viewer"
+
+// ServiceRole specification contains a list of access rules (permissions).
+// This represents the "Spec" part of the ServiceRole object. The name and namespace
+// of the ServiceRole is specified in "metadata" section of the ServiceRole object.
+message ServiceRole {
+  // Required. The set of access rules (permissions) that the role has.
+  repeated AccessRule rules = 1 [(validate.rules).repeated .min_items = 1];
+}
+
+// AccessRule defines a permission to access a list of services.
+message AccessRule {
+  // Required. A list of service names.
+  // Exact match, prefix match, and suffix match are supported for service names.
+  // For example, the service name "bookstore.mtv.svc.cluster.local" matches
+  // "bookstore.mtv.svc.cluster.local" (exact match), or "bookstore*" (prefix match),
+  // or "*.mtv.svc.cluster.local" (suffix match).
+  // If set to ["*"], it refers to all services in the namespace.
+  repeated string services = 1 [(validate.rules).repeated .min_items = 1];
+
+  // Optional. A list of HTTP paths or gRPC methods.
+  // gRPC methods must be presented as fully-qualified name in the form of
+  // packageName.serviceName/methodName.
+  // Exact match, prefix match, and suffix match are supported for paths.
+  // For example, the path "/books/review" matches
+  // "/books/review" (exact match), or "/books/*" (prefix match),
+  // or "*/review" (suffix match).
+  // If not specified, it applies to any path.
+  repeated string paths = 2;
```

Comment:
Do you want to perhaps use a `RouteMatch` instead? https://github.com/envoyproxy/data-plane-api/blob/f88047c0d470352ad1bf6a97dd64155bc65047a0/envoy/api/v2/route/route.proto#L209

Reply:
The match applies to "services", "paths", "constraints" values. I added a StringMatch type, which is similar to RouteMatch.